src/gtc/passes/oir_optimizations/utils.py

```diff
@@ -161,7 +162,7 @@ def symbol_name_creator(used_names: Set[str]) -> Callable[[str], str]:
 
     def increment_string_suffix(s: str) -> str:
         if not s[-1].isnumeric():
-            return s + "0"
+            return s + "_gen_0"
```

Comment:
This is a bit clearer name to grep for in the generated code. Sometimes frontend users would put digits after the name and it became unclear.

Reply:
Maybe even replace the if by `not re.match(r".+_gen_[0-9]+")` to have a stricter distinction between generated symbols and user symbols?

Reply:
Oops, yes I meant to add something like that before. Thanks for catching that!